pages/index.js

```diff
@@ -2,13 +2,37 @@ import useSWR from "swr";
 import styled from "styled-components";
 import AccountBalance from "@/components/AccountBalance";
 import TransactionItem from "@/components/TransactionItem";
+import Form from "@/components/CreateTransaction";
+
 
 export default function HomePage() {
-  const { data: transactions, error, isLoading } = useSWR("/api/transactions");
+  const {
+    data: transactions,
+    error,
+    isLoading,
+    mutate,
+  } = useSWR("/api/transactions");
 
   if (error) return <div>failed to load</div>;
   if (isLoading) return <p>is Loading...</p>;
 
+  async function handleSubmit(formData) {
+    const response = await fetch("/api/transactions", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(formData),
+    });
+
+    if (!response.ok) {
+      console.error("POST failed");
+      return;
+    }
+
+    await response.json();
+    await mutate();
```

Comment:
if you don't need the json response you can omit line 31, also you don't need to await the mutate.